R/template.R

```diff
@@ -77,6 +73,244 @@ metric_summarizer <- function(metric_nm,
                               case_weights = NULL,
                               ...,
                               metric_fn_options = list()) {
+  signal_soft_deprecated(
+    paste(
+      "`metric_summarizer()` has been soft-deprecated as of version 1.2.0.",
+      "Please see `?metric_summarizer()` for further instructions."
+      )
+  )
+  truth <- enquo(truth)
+  estimate <- enquo(estimate)
+  case_weights <- enquo(case_weights)
+
+  validate_not_missing(truth, "truth")
+  validate_not_missing(estimate, "estimate")
+
+  # Explicit handling of length 1 character vectors as column names
+  nms <- colnames(data)
+  truth <- handle_chr_names(truth, nms)
+  estimate <- handle_chr_names(estimate, nms)
+
+  finalize_estimator_expr <- rlang::expr(
+    finalize_estimator(!! truth, estimator, metric_nm)
+  )
+
+  metric_tbl <- dplyr::summarise(
+    data,
+    .metric = metric_nm,
+    .estimator = eval_tidy(finalize_estimator_expr),
+    .estimate = metric_fn(
+      truth = !! truth,
+      estimate = !! estimate,
+      !!! spliceable_estimator(estimator),
+      na_rm = na_rm,
+      !!! spliceable_event_level(event_level),
+      !!! spliceable_case_weights(case_weights),
+      !!! metric_fn_options
+    )
+  )
+
+  dplyr::as_tibble(metric_tbl)
+}
+
+#' Developer function for summarizing new numeric metrics
+#'
+#' `metric_summarizer_numeric()` is useful alongside [metric_vec_template()] for
+#' implementing new custom metrics. `metric_summarizer_numeric()` calls the
+#' metric function inside `dplyr::summarise()`. `metric_vec_template()` is a
+#' generalized function that calls the core implementation of a metric function,
+#' and includes a number of checks on the types, lengths, and argument inputs.
+#' See [Custom performance
+#' metrics](https://www.tidymodels.org/learn/develop/metrics/) for more
+#' information.
+#'
+#' @details
+#'
+#' `metric_summarizer_numeric()` is generally called from the data frame version
+#' of your metric function. It knows how to call your metric over grouped data
+#' frames and returns a `tibble` consistent with other metrics.
+#'
+#'
```

Comment:
I actually think you should try and shove all 3 on the same help documentation page. I think that will help you keep them somewhat in sync going forward.

You won't have to worry about `For numeric metrics, this is left as NULL` because the numeric metric function doesn't have `estimate` or `event_level` as arguments, so you will be able to just document those arguments without mentioning numeric metrics at all

Reply:
Done in https://github.com/tidymodels/yardstick/pull/322/commits/176ea571311c2ab08295dcab83c66c7dca082212